include/kernel.h

```diff
@@ -3550,6 +3550,11 @@ struct k_mem_pool {
 	_MPOOL_HAVE_LVL(maxsz, minsz, 14) +	\
 	_MPOOL_HAVE_LVL(maxsz, minsz, 15))
 
+#define _MPOOL_MINBLK sizeof(sys_dnode_t)
+
+#define _MPOOL_LVLS(max, min)		\
+	__MPOOL_LVLS((max), (min) >= _MPOOL_MINBLK ? (min) : _MPOOL_MINBLK)
```

Comment:
At first glance I wonder if it might be better to fail an assertion or something, instead of just increasing the block size passed in.
You know the mem pool code far better than me though, so I defer to your opinion.